articles/mysql-with-node-js/index.md

```diff
@@ -0,0 +1,598 @@
+
+Once installed, create a project folder and initialize the project with `npm init -y`. Here’s a [guide](/engineering-education/beginner-guide-to-npm) that will help you understand NPM in detail.
+
+To connect Node.js with an SQL database, you need the MySQL package installed.
+
+Run `npm install mysql` to install the package. Create a `.js` file to work on, for example, `app.js`.
+
+### Creating a connection to MySQL server
+Before carrying out any database operation, you first need to create a connection to the MySQL server.
+
+In just a few steps below, and you have a connection to the database established.
+
+- Import MySQL using the `require()` function.
+
+```js
+let mysql = require("mysql");
+```
+
+- Create a connection object with `createConnection()`.
+
+The object specifies your database server's details, such as password, username, and host. In this case, I am using the default values provided by the MySQL server.
+
+```js
+let dbconnection = mysql.createConnection({
+    host: "localhost",
+    user: "root",
+    //MySQL root password
+    password: "",
+});
+```
+
+- Establishing a connection to the server.
+
+To make a connection, use the variable `dbconnection` we have created above and call the `connect()` function.
+
+```js
+dbconnection.connect((err) => {
+    //check connection errors
+    if (err) {
+        return console.error("ERROR: " + err.message);
+    }
+    // execute queries related to the database after a successful connection
+    console.log("connection established...");
+});
+```
+
+The `connect()` function allows us to execute the database queries whenever a connection to the server is established. If a connection was not successful, the server would log an error that describes what went wrong while establishing the connection.
+
+Run `node app.js` to check if the connection is established.
+
+![](/engineering-education/mysql-with-node-js/connection-established.png)
+
+If you run into an error close to the one shown below.
+
+![](/engineering-education/mysql-with-node-js/error.jpg)
+
+Follow [the following steps](https://stackoverflow.com/questions/50093144/mysql-8-0-client-does-not-support-authentication-protocol-requested-by-server).
+
+- Run the following query in MySQL Workbench.
+
+```bash
+ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password'
+```
+
+Where `root` is the user, `localhost` is the URL and `password` is the password. In this case, the password you provided earlier will be overridden, and thus your new password will be `password`.
+
+- Run `flush privileges` to refresh privileges.
+
+We need to create a database that will be queried by the connection we have created.
+
+### Creating a database
+We need to create a database that we will store our information. Create a function `createDB()`. We will use this function to create a new database. After making a successful connection, call the function `createDB()` as shown below.
```

Comment:
Remove `we` in `We need to create a database that we will store our information`.